### DISCUSSION TITLE
flutter

### DISCUSSION THREAD

**Discussion by rraammiinn** [imba/imba#830] [Ideas]
there is a python project called flet that aims to bring flutter to python .
is it possible for imba too ?

**Comment by Bohooslav** [comment]
Anything is possible how think twice:
1. Who is gonna do this, who will maintain it?
2. Who is gonna use it?
3. Will it provide enough features/flexibility versus performance?
4. Is it gonna be really useful and powerful?
5. Is it worth any time and investments?
6. Think again, what do you trade for the syntax sugar you pursue.

**Reply by rraammiinn** [reply]
they previously had typescript support in their roadmap but they seem abandoned it so I thought imba community could do it .



I really like flutter infrastructure but I think gui app development should be something like imba .


performance however is so tricky and I think the best way for achieving performance is that imba compiles to dart but it it isn't good because you will lose js compatibility and all of its libraries and beside it another backed for imba is a really hard work and I don't think anyone do it so the best way seems what flet did but of course you will lose performance .

**Comment by rraammiinn** [comment]
I know that it's irrelevant to imba discussion but what do you think about dart ?
is it a good language ?
will it remains ?

**Comment by Bohooslav** [comment]
Dart has a completely different purpose than Imba. Dart is very old school in syntax and it is called to be cross-platform. While Imba focuses fully on web and syntactic sugar to speed up both performance and development time, easing developer efforts.

Personally, I hate where the idea of cross-platform got so far. I have been working with Flutter and React Native, and I feel very shitty about them, especially about React Native. Web is cross-platform already, but in order to achieve ideal world harmony and peace we need motherfuckers like Apple to comply with standards, which is hard to imagine. 

You personally may give it a shot and try to play around transcompilation, it is your business your time, but there would be more profit if one would spend their time in court with Apple.

**Reply by rraammiinn** [reply]
thanks , I wish they take beauty of programming languages more serious .

unfortunately I don't know anything about language designing .

**Reply by Bohooslav** [reply]
I would recommend you this book if you are interested https://craftinginterpreters.com/

**Reply by rraammiinn** [reply]
thanks , programming languages are very important to me .
